DEVELOPER_GUIDE.md

```diff
@@ -0,0 +1,283 @@
+# Harvester Developer Guide
+
+This guide provides essential information for developers who want to contribute to Harvester.
+
+## Table of Contents
+
+- [Things You Need to Know Before Development](#things-you-need-to-know-before-development)
+  - [Understanding Harvester](#understanding-harvester)
+  - [Rules](#rules)
+- [Repository Relations](#repository-relations)
+  - [Main Components](#main-components)
+  - [Add-ons](#add-ons)
+  - [Rancher-Related](#rancher-related)
+  - [Most Closely Integrated Upstream Repositories](#most-closely-integrated-upstream-repositories)
+- [Development](#development)
+  - [Prerequisites](#prerequisites)
+  - [Basic Knowledge](#basic-knowledge)
+  - [Writing Test Cases](#writing-test-cases)
+- [Debugging and Troubleshooting](#debugging-and-troubleshooting)
+- [Testing and Building](#testing-and-building)
+  - [Test Your Changes by Patching the Image](#test-your-changes-by-patching-the-image)
+  - [Test Your Changes with a Fresh ISO](#test-your-changes-with-a-fresh-iso)
+  - [Automation Testing](#automation-testing)
+- [Before Opening a Pull Request](#before-opening-a-pull-request)
+  - [How to Find an Issue to Work On](#how-to-find-an-issue-to-work-on)
+  - [Branch Strategy](#branch-strategy)
+  - [Code Style](#code-style)
+  - [Commit Message Format](#commit-message-format)
+- [After Opening a Pull Request](#after-opening-a-pull-request)
+- [After All PRs Are Merged](#after-all-prs-are-merged)
+- [Example Issues](#example-issues)
+
+## Things You Need to Know Before Development
+
+### Understanding Harvester
+
+We recommend installing a Harvester cluster to understand how to use its features. We provide several installation methods:
+
+- [ISO Installation](https://docs.harvesterhci.io/v1.6/install/index)
+- [USB Installation](https://docs.harvesterhci.io/v1.6/install/usb-install/)
+- [PXE Boot Installation](https://docs.harvesterhci.io/v1.6/install/pxe-boot-install/)
+- [Net Install ISO](https://docs.harvesterhci.io/v1.6/install/net-install/)
+
+### Rules
+
+Code changes must be submitted via GitHub pull requests. Here are some general guidelines:
+
+- Find or create an issue first. Every PR should link to at least one issue.
+- Fork the repository to which you want to contribute. Make and commit changes in your fork.
+- Create a pull request that targets the appropriate branch, using the PR description template.
+- Sign off your commits by including a Signed-off-by line in each commit message.
+
+## Repository Relations
+
+This section describes the scope of each repository:
+
+### Main Components
```

Comment:
we already have a page list to all components, perhaps we can provide the link here? https://github.com/harvester/harvester/wiki/Component-Maintainers This gives developers an overview of all existing components.